Add ability to retrieve application version info

dan-palmer opened the pull request:
It’s useful across the environments to be able to see what has been
deployed and when it was deployed.

The information returned is consistent with what other apps in the
department return.

Example response from dev:

```json
{
  "build_date":"2017-04-11T14:31:18+0000",
  "build_tag":"pingendpoint.9a6ab35",
  "commit":"9a6ab3526cbf4b26869be7042398682b6a40b241"
}
```

Commit:
Add ability to retrieve application version info

It’s useful across the environments to be able to see what has been
deployed and when it was deployed.

The information returned is consistent with what other apps in the
department return.

With regard to the environment variables we expect here(e.g APP_GIT_COMMIT)
they have been lifted from the ci-deploy repo.

lostie commented on spec/requests/ping_spec.rb:
```diff
@@ -0,0 +1,33 @@
+  describe 'GET /ping' do
+    context 'when the correct environment variables have been set during the deploy process' do
+      before do
+        allow(ENV).to receive(:fetch).with('APP_BUILD_DATE').and_return('2017-04-06T14:47:13+0000')
```
**[not a blocker]** 💡 I've come across this [gem](https://github.com/thoughtbot/climate_control) that achieves the same thing in a slightly cleaner and more [controlled](https://github.com/thoughtbot/climate_control#why-use-climate-control) away.

For this specific case it can be used with a _RSpec around_ block

lostie commented on app/controllers/ping_controller.rb:
```diff
@@ -0,0 +1,17 @@
+
+  def version_info
+    {
+      build_date: ENV.fetch('APP_BUILD_DATE'),
```
Is there any documentation that describes how/when this variables are set?

dan-palmer replied:
@lostie it's a commonly used across projects and is created as part of the build process. I cant find any documentation though - have requested info from platforms. Will update the commit message when they respond.

Commit:
Use ClimateControl to modify the ENV under test

Provides a way to manage the ENV under test. It's a bit better than
stubbing the method we use to request information from the environment,
if we choose to use the bracket accessors instead of fetch the spec will
still pass.
